components/provisioner/internal/templates/render.go

```diff
@@ -0,0 +1,29 @@
+package templates
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+)
+
+type Values struct {
+	ShootName          string
+	ProjectName        string
+	GardenerSecretName string
+	Region             string
+}
+
+func RenderTemplate(text string, values Values) ([]byte, error) {
+	templ, err := template.New("").Parse(text)
+	if err != nil {
+		return nil, fmt.Errorf("error while parsing renderTemplate: %s", err.Error())
+	}
+
+	buffer := new(bytes.Buffer)
+	err = templ.Execute(buffer, values)
+	if err != nil {
+		return nil, fmt.Errorf("error while rendering renderTemplate: %s", err.Error())
```

Comment:
I'm not sure about meaning of this error message. Did you mean (`error while rendering template '%s'`, text)
```suggestion
		return nil, fmt.Errorf("error while rendering template '%s': %s", text, err.Error())
```
?

Reply:
Good catch. Did a little too much renaming, I will fix it with the next changes.